search/includes/classes/class-search.php

```diff
@@ -771,10 +780,37 @@ public function filter__ep_do_intercept_request( $request, $query, $args ) {
 			// Return a generic VIP Search WP_Error instead of the one from wp_remote_request
 			return new \WP_Error( 'vip-search-upstream-request-failed', 'There was an error connecting to the upstream search server' );
 		} else {
+			if ( $is_cacheable ) {
+				// phpcs:ignore WordPress.WP.AlternativeFunctions.rand_mt_rand
+				wp_cache_set( $cache_key, $response, self::SEARCH_CACHE_GROUP, mt_rand( 60, 70 ) );
```

Comment:
Are there certain responses that we don't want to cache?

Reply:
@rebeccahum I think we only want to cache the ones that are querying for or fetching documents. I don't think we want to cache results of indexing requests or DELETEs.

Reply:
Is there a chance that the response could return < 400 code or something and if so, do we want to return that?

Reply:
What if the response is a < 400 error code? Do we want to cache that?

Reply:
I think we may, but I don't have a strong opinion. We can probably cache it for a few seconds but I'm not sure what positive impact that would have.